File: ABC/401-450/401/b.cpp
#include <iostream>
using namespace std;

int main(){
  int n;
  cin >> n;
  bool login = false;
  int ans = 0;
  for(int i=0;i<n;i++){
    string s;
    cin >> s;
    if(s == "login") login = true;
    else if(s == "logout") login = false;
    else if(s == "public" ) continue;
    else if(s == "private"){
      if(!login) ans++;
    }
  }
  cout << ans << endl;
  return 0;
}

File: ABC/401-450/401/b.py
def main():
  n = int(input())
  login = False
  ans = 0
  for i in range(n):
    s = input()
    if s == "login":
      login = True
    elif s == "logout":
      login = False
    elif s == "private" and not login:
      ans += 1
  print(ans)
if __name__ == '__main__':
  main()


